core/app/models/spree/variant/vat_price_generator.rb

```diff
@@ -0,0 +1,58 @@
+module Spree
+  class Variant
+    # This class generates gross prices for all countries that have VAT configured.
+    # The prices will include their respective VAT rates. It will also generate an
+    # export (net) price for any country that doesn't have VAT.
+    # @example
+    #   The admin is configured to show German gross prices
+    #   (Spree::Config.admin_vat_country_iso == "DE")
+    #
+    #   There is VATs configured for Germany (19%) and Finland (24%).
+    #   The VAT price generator is run on a variant with a base (German) price of 10.00.
+    #   The Price Generator will generate:
+    #     - A price for Finland (country_id == "FI"): 10.42
+    #     - A price for any other country (country_id == nil): 8.40
+    #
+    class VatPriceGenerator
+      attr_reader :variant
+
+      def initialize(variant)
+        @variant = variant
+      end
+
+      def run
+        # Early return if there is no VAT rates in the current store.
+        return if !variant.tax_category || variant_vat_rates.empty?
+
+        country_isos_requiring_price.each do |country_iso|
+          # Don't re-create the default price
+          next if variant.default_price && variant.default_price.country_iso == country_iso
+
+          foreign_price = variant.prices.find_or_initialize_by(
+            country_iso: country_iso,
+            currency: variant.default_price.currency,
+          )
+
+          foreign_price.amount = variant.default_price.net_amount * (1 + vat_for_country_iso(country_iso))
+        end
+      end
+
+      private
+
+      # nil is added to the array so we always have an export price.
+      def country_isos_requiring_price
+        return [nil] unless variant.tax_category
+        [nil] + variant_vat_rates.map(&:zone).flat_map(&:countries).flat_map(&:iso)
```

Comment:
An export price is a price without VAT, as in countries with VAT, if you export to a country where no VAT applies, you have to change the price to exclude VAT. We add `nil` because a price with `nil` is used for shipping to any country we have no information about. Otherwise we'd need to create a price for each country, which is redundant and tedious to work with.